### ISSUE TITLE
Support OAuth 2.0

### ISSUE THREAD

**Issue by diegodlh** [diegodlh/zotero-wikicite#46]
The plugin currently asks for username and password to login to Wikidata each time an edit batch is requested. Eventually, these may be saved to local storage if desired by the user (#45).

Ideally, plugin should support logging in using OAuth. However, OAuth 2 would be needed, because the client is running on the user's computer and can't keep OAuth 1's consumer secret safe.

OAuth 2 is [not yet supported](https://github.com/maxlath/wikibase-edit/issues/50) by wikibase-edit.

**Comment by diegodlh** [comment]
Refer to MediaWiki's OAuth 2 documentation for developers: https://www.mediawiki.org/wiki/OAuth/For_Developers#Authorization_2

**Comment by Poslovitch** [comment]
`wikibase-edit` seems to support OAuth 1.0a, but still not 2.0 however. (see https://codeberg.org/maxlath/wikibase-edit/issues/79).

I'll try to contribute to it.

**Comment by Dominic-DallOsto** [comment]
> `wikibase-edit` seems to support OAuth 1.0a, but still not 2.0 however. (see https://codeberg.org/maxlath/wikibase-edit/issues/79).
> 
> I'll try to contribute to it. 

Yeah exactly. If you manage to get OAuth 2.0 working with wikibase-edit that would be amazing!

**Comment by Poslovitch** [comment]
`wikibase-edit` should now support OAuth2 as of release `8.2.0` as per the following ticket: https://codeberg.org/maxlath/wikibase-edit/issues/79#issuecomment-20216899.

maxlath was quicker than me at implementing this, I'm still recovering from Wikimania 2026.
Hope it helps :)

**Comment by Dominic-DallOsto** [comment]
This was added in 2b773d67b357d085c90678df08846aea8245d521 (see #70)

But the process could still be refined so I'll leave this issue open for now

**Comment by Dominic-DallOsto** [comment]
Quickstatements itself seems to have a nice example of how to do the oauth login smoothly (it redirects you to login at mediawiki) - see here https://github.com/magnusmanske/quickstatements/blob/master/public_html/quickstatements.php#L149

I think quickstatements v3 let you log in via wikidata itself, which might be even nicer
